plugins/mavlink_backdoor/include/plugins/mavlink_backdoor/mavlink_backdoor.h

```diff
@@ -0,0 +1,125 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <functional>
+
+// This plugin provides/includes the mavlink 2.0 header files.
+#include "mavlink/v2.0/common/mavlink.h"
+#include <dronecode_sdk/plugin_base.h>
+
+namespace dronecode_sdk {
+
+class System;
+class MavlinkBackdoorImpl;
+
+/**
+ * @brief The MavlinkBackdoor class provides direct MAVLink access
+ *
+ * "With great power comes great responsibility." - This plugin allows
+ * you to send and receive MAVLink messages. There is no checking or
+ * safe-guards, you're on your own, and you have been warned.
+ */
+class MavlinkBackdoor : public PluginBase {
+public:
+    /**
+     * @brief Constructor. Creates the plugin for a specific System.
+     *
+     * The plugin is typically created as shown below:
+     *
+     *     ```cpp
+     *     auto mavlink_backdoor = std::make_shared<MavlinkBackdoor>(system);
+     *     ```
+     *
+     * @param system The specific system associated with this plugin.
+     */
+    explicit MavlinkBackdoor(System &system);
+
+    /**
+     * @brief Destructor (internal use only).
+     */
+    ~MavlinkBackdoor();
+
+    /**
+     * @brief Possible results returned for requests.
+     */
+    enum class Result {
+        UNKNOWN, /**< @brief Unknown error. */
+        SUCCESS, /**< @brief Success. */
+        CONNECTION_ERROR /**< @brief Connection error. */
+    };
+
+    /**
+     * @brief Returns a human-readable English string for `MavlinkBackdoor::Result`.
+     *
+     * @param result The enum value for which a human readable string is required.
+     * @return Human readable string for the `MavlinkBackdoor::Result`.
+     */
+    std::string result_str(Result result);
+
+    /**
+     * @brief Send message.
+     *
+     * @return result of the request.
+     */
+    Result send_message(const mavlink_message_t &message);
+
+    /**
+     * @brief Subscribe to messages using message ID.
+     *
+     * This means that all future messages being received will trigger the
+     * callback to be called. To stop the subscription, call this method with
+     * `nullptr` as the argument.
+     *
+     * @param message_id The MAVLink message ID.
+     * @param callback Callback to be called for message subscription.
```

Comment:
I would like this to be more generic:
- A callback for all received messages independent from the ID
- Ability to modify the message
- The callback should return a bool: if false, the message will be dropped.

These are requirements for the testing suite.

Reply:
> - Ability to modify the message

Modify before it goes to other plugins? Ok, that should be possible to add.

> - The callback should return a bool: if false, the message will be dropped.

I don't get it. Dropped from where? Ah, so we need a hook that is called before any other plugins (or the core) get the message delivered.

> These are requirements for the testing suite.

I was planning to come up with higher level hooks for testing such as "drop 50% of the incoming messages" but that might not yield consistent results, so I need to think a bit more about that.